lib/storage/storage.go

```diff
@@ -1027,18 +1027,23 @@ func SetRetentionTimezoneOffset(offset time.Duration) {
 var retentionTimezoneOffsetMsecs int64
 
 func nextRetentionDuration(retentionMsecs int64) time.Duration {
-	// Round retentionMsecs to days. This guarantees that per-day inverted index works as expected.
-	retentionMsecs = ((retentionMsecs + msecPerDay - 1) / msecPerDay) * msecPerDay
```

Comment:
`indexdb` rotation cannot be performed with the frequency smaller than one day because the `indexdb` stores per-day index. It would be great to add tests for this function for retentions close to one day and make sure that it works as expected.